processing/src/main/java/org/apache/druid/segment/virtual/SingleLongInputCachingExpressionColumnValueSelector.java

```diff
@@ -155,4 +130,26 @@ public boolean isNull()
     // ExprEval.isNumericNull checks whether the parsed primitive value is null or not.
     return getObject().isNumericNull();
   }
+
+  public class LruEvalCache
+  {
+    private final Long2ObjectLinkedOpenHashMap<ExprEval> m = new Long2ObjectLinkedOpenHashMap<>(CACHE_SIZE);
```

Comment:
Maybe don't size it from the beginning and let it grow

Reply:
Sounds like a good idea. I changed it.